pkg/kubectl/cmd/convert.go

```diff
@@ -254,16 +254,21 @@ func asVersionedObject(infos []*resource.Info, forceList bool, version schema.Gr
 func asVersionedObjects(infos []*resource.Info, version schema.GroupVersion, encoder runtime.Encoder) ([]runtime.Object, error) {
 	objects := []runtime.Object{}
 	for _, info := range infos {
-		if info.Object == nil {
+		var obj runtime.Object
+		if info.VersionedObject != nil {
+			obj = info.VersionedObject
```

Comment:
there's no guarantee the versioned object is convertible to the desired output. It usually isn't.